pandas/_libs/lib.pyx

```diff
@@ -1171,6 +1172,9 @@ def infer_dtype(object value, bint skipna=False):
         values = construct_1d_object_array_from_listlike(value)
 
     values = getattr(values, 'values', values)
+    if skipna:
+        values = values[~isnaobj(values)]
+
```

Comment:
Unfortunately, that's not directly possible (nor performant), because the line directly below (with `_try_infer_map`) will return prematurely as soon as it can grab hold of a dtype.